### PR TITLE
add auth from auth0

### PR DESCRIPTION
**Pull request by Antik88**

### CODE REVIEW

File: User.Service/User.Service.API/Controllers/UserController.cs [hunk 1]
```diff
@@ -11,24 +12,23 @@ namespace User.Service.API.Controllers;
...
 
         var newUserModel = await userService.AddAsync(userModel, cancellationToken);
 
-        var newUserDto = mapper.Map<CreateUserDto>(newUserModel);
+        var newUserDto = mapper.Map<UserDto>(newUserModel);
 
         return newUserDto;
     }
 
+    [Authorize (Roles = "Admin")]
```
Comment (moltenessence) [inline User.Service/User.Service.API/Controllers/UserController.cs]: could be extracted to constants

File: User.Service/User.Service.API/Controllers/UserController.cs [hunk 2]
```diff
@@ -44,6 +44,7 @@ public class UserController(
         };
     }
 
+    [Authorize (Roles = "User")]
```
Comment (moltenessence) [inline User.Service/User.Service.API/Controllers/UserController.cs]: same

File: User.Service/User.Service.API/Controllers/UserController.cs [hunk 3]
```diff
@@ -54,14 +55,16 @@ public class UserController(
...
     {
         return userService.Delete(id, cancellationToken);
     }
 
+    [Authorize]
     [HttpPatch("{id}")]
-    public async Task<CreateUserDto> DeleteById(
+    public async Task<CreateUserDto> PatchById(
         [FromRoute] Guid id,
         [FromBody] CreateUserDto userDto,
```
Comment (moltenessence) [inline User.Service/User.Service.API/Controllers/UserController.cs]: why you use CreateUserDto in patch method (and return this type as well)?

File: User.Service/User.Service.BLL/Models/UserModel.cs [hunk 4]
```diff
@@ -3,9 +3,8 @@
...
     public Guid Id { get; set; }
-    public string? AuthId { get; set; }
-    public string Name { get; set; }
-    public string Email { get; set; }
-    public string Password { get; set; }
-    public string Connection { get; set; }
+    public string AuthId { get; set; } = string.Empty;
+    public string Name { get; set; } = string.Empty;
+    public string Email { get; set; }  = string.Empty;
+    public string? UserPicture { get; set; }
```
Comment (ZaytsevGleb) [inline User.Service/User.Service.BLL/Models/UserModel.cs]: Why userPicture is string?

File: User.Service/User.Service.API/DI/ServicesConfiguration.cs [hunk 5]
```diff
@@ -33,6 +33,11 @@ private static void AddAuth0Authentication(this IServiceCollection services, ICo
         {
             options.Authority = $"https://{configuration["Auth0:Domain"]}/";
```
Comment (ZaytsevGleb) [inline User.Service/User.Service.API/DI/ServicesConfiguration.cs]: U can move full url to configuration

File: User.Service/User.Service.API/Program.cs [hunk 6]
```diff
@@ -26,7 +26,10 @@ public static void Main(string[] args)
 
         app.UseHttpsRedirection();
 
-        // Ensure authentication and authorization middleware are included in the pipeline
+        app.UseCors(builder =>
+              builder.WithOrigins("http://localhost:5173")
```
Comment (5Ka-me) [inline User.Service/User.Service.API/Program.cs]: Move this const into the `appsettings`

**Review by 5Ka-me — CHANGES_REQUESTED** [review]

**Review by ZaytsevGleb — APPROVED** [review]

**Review by pavelGagarin0870 — APPROVED** [review]

**Review by 5Ka-me — APPROVED** [review]

**Pull request merged**